server/container/util-java8/src/main/java/org/apache/james/util/MissableOptional.java

```diff
@@ -19,82 +19,124 @@
 
 package org.apache.james.util;
 
+import java.util.NoSuchElementException;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.function.Supplier;
 
+import com.google.common.base.Preconditions;
+
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
```

Comment:
Done. I replaced by a runtime exception as Apache NotImplementedException is not available here.

I prefer to keep dependencies low as it used pretty every where.